### ISSUE TITLE
How to modify redis.conf

### ISSUE THREAD

**Issue by AhaYujie** [Grokzen/docker-redis-cluster#76]
Is there a way to modify redis.conf.
I see **redis.tmpl**  file, is it redis.conf?

**Comment by Grokzen** [comment]
It should be possible to modify both redis-cluster.tmpl and redis.tmpl, then rebuild the container locally to inject your modifications and then start the container with including your modifications. There is no option to inject config options if you are just downloading the image from docker-hub.

**Comment by AhaYujie** [comment]
I rebuild  the container after I modify redis.tmpl like this.
```
#bind 0.0.0.0
 protected-mode no
 port ${PORT}
 appendonly yes
 dir /redis-data/${PORT}
```
I enter inside the container and check redis.conf
```
docker exec -it redis-cluster-container bash
cd /redis-conf
cat redis.tmpl
```
and get
```
#bind 0.0.0.0
protected-mode no
port ${PORT}
appendonly yes
dir /redis-data/${PORT}
```
but I check other info
```
cd /redis-conf/7000
cat redis.conf
```
get 
```
bind 0.0.0.0
port 7000
cluster-enabled yes
cluster-config-file nodes.conf
cluster-node-timeout 5000
appendonly yes
dir /redis-data/7000
```
and I enter redis and check
```
redis-cli -c -p 7000
config get bind
```
get 
```
1) "bind"
2) "0.0.0.0"
```
which is not working as I exepected.
I did something wrong? or modify redis.tmpl just dont work?

**Comment by AhaYujie** [comment]
It is not working by changing redis.tmpl

**Comment by ericsantanna** [comment]
It's a closed an old question but I had the same problem.

Solved it by editing the Dockerfile.

From:
`ENTRYPOINT ["/docker-entrypoint.sh"]`
To:
`ENTRYPOINT ["/docker-entrypoint.sh", "redis-cluster"]`

**Comment by AlexisWilke** [comment]
Using a docker-compose.yaml, I was able to use the default docker and change the redis configuration by defining a volumes entry. Something like this:

1. create a local `redis.conf` file

    bind ${BIND_ADDRESS}
    port ${PORT}
    appendonly yes
    dir /redis-data/${PORT}
    protected-mode no

2. edit the `docker-compose.yaml` file

      redis-standalone:
        ...
        volumes:
          - ./redis.conf:/redis-conf/redis.tmpl

To create the file in step 1, I just looked at the existing one from the Docker Desktop terminal:

    cd /redis-conf
    cat redis.impl

then added the `protected-mode no` at the end.

Note that the filename uses `.tmpl`, not `.conf`.

Similarly, you want to do that for the `redis-cluster.tmpl` file for a cluster setup.

And that was sufficient to get the flag in the docker without having to rebuild the docker itself.
